### PR TITLE
Feat/early contribution

### PR DESCRIPTION
**Pull request by GopherJ**
close: #968

### CODE REVIEW

**Comment by GopherJ** [comment]
do you have better idea of `early contribution` @ETeissonniere ? now we use pending to record what haven't been contributed on relaychain at the stage `Pending`

After calling `open` it enters the phase `Contributing` then the next contribute call will bring all all `pending` to relaychain and clean `pending`

File: pallets/crowdloans/src/lib.rs [hunk 1]
```diff
@@ -268,6 +270,28 @@ pub mod pallet {
...
+            Vaults::<T>::try_mutate(&crowdloan, |vault| -> Result<_, DispatchError> {
+                let mut vault = vault.as_mut().ok_or(Error::<T>::VaultDoesNotExist)?;
+
+                ensure!(
+                    vault.phase == VaultPhase::Pending,
+                    Error::<T>::IncorrectVaultPhase
+                );
+
+                vault.phase = VaultPhase::Contributing;
+
```
Comment (ETeissonniere) [inline pallets/crowdloans/src/lib.rs:288]: Shouldn't we call `do_contribute` here and update the state accordingly?

**Comment by ETeissonniere** [comment]
@GopherJ I like the way it is currently, shall we just auto contribute what is pending during the `open` call?

**Comment by GopherJ** [comment]
> 

Yes I think it's a good idea!

**Pull request merged**